core-tests/shared/src/test/scala/zio/ZManagedSpec.scala

```diff
@@ -1208,7 +1208,6 @@ object ZManagedSpec extends ZIOBaseSpec {
           releaseLatch <- Promise.make[Nothing, Unit]
           managed       = ZManaged.reserve(Reservation(reserveLatch.await, _ => releaseLatch.succeed(())))
           res          <- managed.timeout(Duration.Zero).use(ZIO.succeed(_))
-          _            <- reserveLatch.succeed(())
```

Comment:
Don't we need this? The contract for `timeout` is that the finalizer will be run if the acquire action completes successfully, even after the timeout. But the acquire can't complete until the latch is completed. If anything it seems problematic that the test sometimes passes without this.

Reply:
I see, make sense. I will go ahead and close this PR.